doc/source/whatsnew/v1.1.0.rst

```diff
@@ -856,6 +856,7 @@ Performance improvements
 - Performance improvement in arithmetic operations (sub, add, mul, div) for MultiIndex (:issue:`34297`)
 - Performance improvement in `DataFrame[bool_indexer]` when `bool_indexer` is a list (:issue:`33924`)
 - Significant performance improvement of :meth:`io.formats.style.Styler.render` with styles added with various ways such as :meth:`io.formats.style.Styler.apply`, :meth:`io.formats.style.Styler.applymap` or :meth:`io.formats.style.Styler.bar` (:issue:`19917`)
+- Performance improvement in the :func:`Series.replace` when `to_replace` is a dict (:issue:`33920`)
```

Comment:
Does this improve performance relative to 1.0.5, or just master? We only need this if it's faster relative to 1.0.5

Reply:
Locally, I have this at ~1.5x slower than 1.0.5 on the example from https://github.com/pandas-dev/pandas/issues/33920. So seems like we could still use some work to get us back to 1.0.x speeds, as long as we don't sacrifice correctness.

Reply:
Yes indeed, those improvements are only relative to master. So, maybe it doesn't make sense to list this change at all in the whatsnew document. I'll have it removed. However, this should still be merged to 1.1.0 (if we manage on time) right?